### PR TITLE
(SIMP-7779) Standardize local rule queue defaults

### PR DESCRIPTION
**Pull request by kendall-moore**
- Update local rule defined type to use the same package defaults for
queues as the remote rule defined type
- Update local rule template to check for optional parameters
- Update spec tests to properly reflect these new defaults

SIMP-7779 #close

### CODE REVIEW

**Comment by kendall-moore** [comment]
GitLab CI Pipeline for reference: https://gitlab.com/simp/pupmod-simp-rsyslog/pipelines/149861446

File: manifests/rule/local.pp [hunk 1]
```diff
@@ -154,29 +154,30 @@
...
   Optional[Integer[0]]           $queue_checkpoint_interval            = undef,
   Boolean                        $queue_sync_queue_files               = false,
   Rsyslog::QueueType             $queue_type                           = 'Direct',
-  Integer[0]                     $queue_worker_threads                 = 1,
-  Integer[0]                     $queue_timeout_shutdown               = 0,
-  Integer[0]                     $queue_timeout_action_completion      = 1000,
-  Integer[0]                     $queue_timeout_enqueue                = 2000,
-  Integer[0]                     $queue_timeout_worker_thread_shutdown = 60000,
-  Integer[0]                     $queue_worker_thread_minimum_messages = 100,
-  String[1]                      $queue_max_file_size                  = simplib::dlookup('rsyslog::rule::local', 'queue_max_file_size', $name, { 'default_value' => '1m' }),
```
Comment (trevor-vaughan) [inline manifests/rule/local.pp:177]: We should leave the `dlookup` here. Should be able to still make it `Optional` and return `undef` by default.

Comment (trevor-vaughan) [reply manifests/rule/local.pp:177]: Bah, I missed the update to metadata.json. We should leave the `dlookup` though

File: manifests/rule/local.pp [hunk 2]
```diff
@@ -154,29 +154,30 @@
...
+  Optional[Integer[0]]           $queue_worker_threads                 = undef,
+  Optional[Integer[0]]           $queue_timeout_shutdown               = undef,
+  Optional[Integer[0]]           $queue_timeout_action_completion      = undef,
+  Optional[Integer[0]]           $queue_timeout_enqueue                = undef,
+  Optional[Integer[0]]           $queue_timeout_worker_thread_shutdown = undef,
+  Optional[Integer[0]]           $queue_worker_thread_minimum_messages = undef,
+  Optional[String[1]]            $queue_max_file_size                  = undef,
```
Comment (trevor-vaughan) [inline manifests/rule/local.pp:178]:
```suggestion
  Optional[String[1]]            $queue_max_file_size                  = simplib::dlookup('rsyslog::rule::local', 'queue_max_file_size', $name, { 'default_value' => '1m' }),
```

File: manifests/rule/local.pp (hunk 1 above)
Comment (kendall-moore) [reply manifests/rule/local.pp:177]: Fixed.

**Comment by kendall-moore** [comment]
@trevor-vaughan I added the dlookup back with a default value of undef. I think we should be good to go now.

**Comment by ayohrling** [comment]
This all seems good to me code-wise. Makes sense with the validations the way they are handled and tests passed locally.

**Comment by trevor-vaughan** [comment]
Travis is having issues reporting again. Using admin powers to submit.

**Pull request merged**